File: FindTheDifference/findTheDifference.py
def findTheDifference(s: str, t: str) -> str:
    res = ""
    t_dict = {}
    for i in t:
        if i in t_dict:
            t_dict[i] += 1 
        else:
            t_dict[i] = 1
    for i in s:
        if i in t_dict:
            t_dict[i] -= 1
        print(t_dict)
    for key, val in t_dict.items():
        if val == 1:
           return key 

s = "abc"
t = "abcd"
print(findTheDifference(s, t))

s = "a"
t = "aa"
print(findTheDifference(s, t))

s = "ae"
t = "aea"
print(findTheDifference(s, t))

File: FindTheDifference/findTheDifference.go
package main

import "fmt"

func findTheDifference(s string, t string) byte {
	s_AsciSum := rune(0)
	for _, val := range s {
		s_AsciSum += val
	}
	t_AsciSum := rune(0)
	for _, val := range t {
		t_AsciSum += val
	}
	res := t_AsciSum - s_AsciSum
	return byte(res)
}

func main() {
	var s, t string

	s = "ab"
	t = "abc"
	fmt.Println(findTheDifference(s, t))

	s = "a"
	t = "aa"
	fmt.Println(findTheDifference(s, t))

	s = "ae"
	t = "aea"
	fmt.Println(findTheDifference(s, t))


}
